app/src/main/java/chat/rocket/android/profile/ui/ProfileFragment.kt

```diff
@@ -66,6 +111,102 @@ class ProfileFragment : Fragment(), ProfileView, ActionMode.Callback {
         listenToChanges()
     }
 
+    override fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray) {
+        when (requestCode) {
+            CHOOSE_PICKER_MODE -> {
+                if (grantResults.size > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+                    openImagePickerChooserDialog()
+                } else {
+                    Toast.makeText(context, getString(R.string.permission_image_picking_not_allowed), Toast.LENGTH_SHORT).show()
+                }
+            }
+        }
+    }
+
+    //show dialog to choose whether to pick image from gallery or to click it from camera
+    private fun openImagePickerChooserDialog() {
+        val choices = arrayOf("Take photo", "Choose photo from storage")
```

Comment:
You need to add a string entry in the strings.xml file.
Also, change `Choose photo from storage` to `Choose photo from gallery`.

Translations for PT-BR: `Tirar foto`. `Escolher foto da galeria`.